infrastructure/src/main/java/org/corfudb/infrastructure/ServerContext.java

```diff
@@ -244,6 +246,34 @@ public ExecutorService getExecutorService(int threadCount, @Nonnull ThreadFactor
         return Executors.newFixedThreadPool(threadCount, threadFactory);
     }
 
+    /**
+     * Get a bounded ExecutorService that can be used by the servers to
+     * process RPCs. Uses a ServerThreadFactory as the underlying
+     * thread factory.
+     * @param threadCount   The number of threads to use in the pool
+     * @param threadPrefix  The naming prefix
+     * @param queueCapacity  Max capacity of underlying linked blocking queue
+     * @return The newly created ExecutorService
+     */
+    public ExecutorService getExecutorService(int threadCount, String threadPrefix, int queueCapacity) {
+        return getBoundedExecutorService(threadCount,
+                new ServerThreadFactory(threadPrefix, new ServerThreadFactory.ExceptionHandler()),
+                queueCapacity);
+    }
+
+    /**
+     * Get a bounded ExecutorService that can be used by the servers to
+     * process RPCs.
+     * @param threadCount    The number of threads to use in the pool
+     * @param threadFactory  The underlying thread factory
+     * @param queueCapacity  Max capacity of underlying linked blocking queue
+     * @return The newly created ExecutorService
+     */
+    public ExecutorService getBoundedExecutorService(int threadCount, @Nonnull ThreadFactory threadFactory, int queueCapacity) {
+        return new ThreadPoolExecutor(threadCount, threadCount, 0L, TimeUnit.MILLISECONDS,
+                new LinkedBlockingQueue<>(queueCapacity), threadFactory, new AbortPolicy());
+    }
+
```

Comment:
This seems like its a little verbose and no generic for reuse. I think its cleaner to just get the number of pending requests in `processRequest` with `((ThreadPoolExecutor)executor).getQueue().size()`